### ISSUE TITLE
Backward-compatible way to add arbitrary lines to sections

### ISSUE THREAD

**Issue by chmduquesne** [Oefenweb/ansible-haproxy#87]
Hi there,

The current config is nice, but has some problems if you have requirements about the order of the lines, or if you need unsupported options.

I was thinking that it would be nice to support inserting arbitrary lines in a section. It could be implemented this way:

- Each of the sections may have a `raw_options` attribute, an array of lines which defaults to [].
- Each section template, expands those `raw_options` like this:
```jinja2
{% for raw_option in <section>.raw_options | default([]) %}
    {{ raw_option }}
{% endfor %}
```

This way, this role would support any configuration, without breaking backward compatibility.

What do you think? Would this be a suitable addition?

Cheers,
@chmduquesne

**Comment by chmduquesne** [comment]
I must highlight that I am aware of the possibility to provide your own template.

It breaks my workflow, since I like to configure everything exclusively through variables. Providing an extra template file requires maintaining 2 files instead of 1: The template, and the variables for the role. I'd rather stick to variables only.

**Comment by tersmitten** [comment]
In the long run I would like to make it more like https://github.com/Oefenweb/ansible-nginx, but for now this seems like a great addition

**Comment by chmduquesne** [comment]
> In the long run I would like to make it more like https://github.com/Oefenweb/ansible-nginx

That would be great!

> but for now this seems like a great addition

Shall I give it a go? Any wish about the implementation? I am thinking that if you want it to look like https://github.com/Oefenweb/ansible-nginx, then maybe I should use a variable named `directives` instead of `raw_options`.

**Comment by tersmitten** [comment]
Fixed in #89

**Comment by chmduquesne** [comment]
Thanks !
